modules/OCPP201/conversions.cpp

```diff
@@ -1251,5 +1251,41 @@ to_everest_set_variable_status_enum_type(const ocpp::v201::SetVariableStatusEnum
     }
 }
 
+types::ocpp::ChargingSchedules to_everest_charging_schedules(const std::vector<ocpp::v201::CompositeSchedule>& composite_schedules) {
+    types::ocpp::ChargingSchedules charging_schedules;
+    for (const auto& composite_schedule : composite_schedules) {
+        charging_schedules.schedules.push_back(conversions::to_everest_charging_schedule(composite_schedule));
+    }
+    return charging_schedules;
+}
+
+types::ocpp::ChargingSchedule to_everest_charging_schedule(const ocpp::v201::CompositeSchedule& composite_schedule) {
+    types::ocpp::ChargingSchedule charging_schedule;
+    charging_schedule.connector =
+        composite_schedule.evseId; // TODO: I think we can remove connector from the EVerest type
+    charging_schedule.charging_rate_unit =
+        ocpp::v201::conversions::charging_rate_unit_enum_to_string(composite_schedule.chargingRateUnit);
+    charging_schedule.evse = composite_schedule.evseId;
+    charging_schedule.duration = composite_schedule.duration;
+    charging_schedule.start_schedule = composite_schedule.scheduleStart.to_rfc3339();
+    // TODO: min_charging_rate ?
+    for (const auto& charging_schedule_period : composite_schedule.chargingSchedulePeriod) {
+        charging_schedule.charging_schedule_period.push_back(
+            to_everest_charging_schedule_period(charging_schedule_period));
+    }
+    return charging_schedule;
+}
+
+types::ocpp::ChargingSchedulePeriod
+to_everest_charging_schedule_period(const ocpp::v201::ChargingSchedulePeriod& period) {
+    types::ocpp::ChargingSchedulePeriod _period;
+    _period.start_period = period.startPeriod;
+    _period.limit = period.limit;
+    _period.stack_level = 0; // TODO: Find a solution for this
```

Comment:
Do you know why a `types::ocpp::ChargingSchedulePeriod` needs a stack level to begin with? It seems like we're creating a leaky abstraction by relying on stack levels outside of `libocpp`.

Reply:
I agree. It is used in some EVerest applications to identify additional sources / purposes (in addition to the purposes that are defined by OCPP) of charging profiles. I saw your comment on that [here](https://github.com/EVerest/libocpp/issues/759) and agree with the approach. I'm open to discuss possible solutions to clean this up in the future.